### PR TITLE
SW-4384 List Nursery Batches by Batch

### PR DESCRIPTION
**Pull request by charlie-tf**
- Create table view for batches_by_batch
- Link batchNumber column to batch details page
- Create method in service to get all batches
  - With nursery filter
  - With search

image

### CODE REVIEW

File: src/components/InventoryV2/InventoryBatch.tsx [hunk 1]
```diff
@@ -133,15 +133,21 @@ export default function InventoryBatch({ origin, species }: InventoryBatchProps)
       <PageHeaderWrapper nextElement={contentRef.current}>
         <BackToLink
           id='back'
-          name={`${strings.INVENTORY} / ${
-            origin === 'Species'
-              ? strings.formatString(strings.BATCHES_OF, getSpeciesLabel())
-              : strings.formatString(strings.BATCHES_AT, getNurseryLabel())
-          } `}
+          name={`${strings.INVENTORY}${
+            origin !== 'Inventory'
```
Comment (charlie-tf) [inline src/components/InventoryV2/InventoryBatch.tsx:137]: When linked from the batches list page, the breadcrumb should not have a species or nursery

**Review by karthikbtf — COMMENTED** [review]
Very nice, a few comments.
Just wanted to call out an addition feature - link from Locations -> Nursery -> sublocations-table to this page. https://terraformation.atlassian.net/browse/SW-4434

File: src/components/InventoryV2/InventoryListByBatch.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        germinatingQuantity: numericFormatter.format(resultTyped['germinatingQuantity(raw)']),
+        notReadyQuantity: numericFormatter.format(resultTyped['notReadyQuantity(raw)']),
+        readyQuantity: numericFormatter.format(resultTyped['readyQuantity(raw)']),
+        totalQuantity: numericFormatter.format(resultTyped['totalQuantity(raw)']),
+      } as InventoryResultWithBatchNumber;
+    });
+
+    if (updatedResult) {
+      if (!debouncedSearchTerm && !filters.facilityIds?.length) {
+        setUnfilteredInventory(updatedResult);
```
[the remark this answers is not in the document]
Comment (charlie-tf) [reply src/components/InventoryV2/InventoryListByBatch.tsx:117]: That would probably be better, yeah. Just some copy-pasta 🍝

File: src/components/InventoryV2/InventoryListByBatch.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+    // format results
+    updatedResult = apiSearchResults?.map((uR) => {
+      const resultTyped = uR as BatchInventoryResult;
+      return {
+        ...resultTyped,
+        batchId: resultTyped.id,
+        species_scientificName_noLink: resultTyped.species_scientificName,
+        facility_name_noLink: resultTyped.facility_name,
+        germinatingQuantity: numericFormatter.format(resultTyped['germinatingQuantity(raw)']),
```
Comment (nickgraz) [inline src/components/InventoryV2/InventoryListByBatch.tsx:108]: I was under the impression that accessing `germinatingQuantity`, without the `(raw)`, part is already localized... is that right? If that's true, would it be OK to just use that?

Comment (karthikbtf) [reply src/components/InventoryV2/InventoryListByBatch.tsx:108]: yes, good catch

Comment (charlie-tf) [reply src/components/InventoryV2/InventoryListByBatch.tsx:108]: Good point, I'll change that. Thanks

**Review by karthikbtf — APPROVED** [review]
LGTM!
Wanted to make sure you saw my earlier comment  '..Just wanted to call out an addition feature - link from Locations -> Nursery -> sublocations-table to this page. https://terraformation.atlassian.net/browse/SW-4434...' (for a future effort perhaps)

**Comment by charlie-tf** [comment]
> LGTM! Wanted to make sure you saw my earlier comment '..Just wanted to call out an addition feature - link from Locations -> Nursery -> sublocations-table to this page. https://terraformation.atlassian.net/browse/SW-4434...' (for a future effort perhaps)

Oh! Sorry I overlooked that bit. I'll create a followup ticket for it 👍

**Pull request merged**